src/sumbuddy/mapper.py

```diff
@@ -39,6 +39,10 @@ def gather_file_paths(self, input_directory, ignore_file=None, include_hidden=Fa
         ---------
         file_paths - List. Files in input_directory that are not ignored.
         """
+
+        if not os.path.isdir(input_directory):
+            raise NotADirectoryError(input_directory)
```

Comment:
It makes more sense to me to raise an error if `gather_file_paths` receives a single file as an input than to repeat its path as an output. This doesn't interfere with use of the CLI or `get_checksums` method with a single file input.